[CI] MultiClusterSpecIT test {stats.MaxOfIpGrouping} failing

**Build Scans:**
- [elasticsearch-intake #8375 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/k2v57luyn7azu)
- [elasticsearch-pull-request #26465 / 8.16.0_bwc-snapshots](https://gradle-enterprise.elastic.co/s/mqxmib7ocagps)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:multi-clusters:v8.16.0#bwcTest" -Dtests.class="org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT" -Dtests.method="test {stats.MinOfIpGrouping}" -Dtests.seed=D2BF9FC1A7C33075 -Dtests.bwc=true -Dtests.locale=no-NO -Dtests.timezone=America/Mexico_City -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Bstats.MaxOfIpGrouping%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Data mismatch:
row 0 column 0:a list containing
row 0 column 0:0: expected "127.0.0.1" but was "fe80::cae2:65ff:fece:feb9"
row 0 column 1:a list containing
row 0 column 1:0: expected "127.0.0.1" but was "fe80::cae2:65ff:fece:feb9"
row 0 column 2:a list containing
row 0 column 2:0: expected "127.0.0.1" but was "fe80::cae2:65ff:fece:feb9"
row 0 column 4:a list containing
row 0 column 4:0: expected "beta" but was "epsilon"
row 1 column 0:a list containing
row 1 column 0:0: expected "fe80::cae2:65ff:fece:feb9" but was "127.0.0.1"
row 1 column 1:a list containing
row 1 column 1:0: expected "fe80::cae2:65ff:fece:feb9" but was "127.0.0.1"
row 1 column 2:a list containing
row 1 column 2:0: expected "fe80::cae2:65ff:fece:feb9" but was "127.0.0.1"
row 1 column 3:a list containing
row 1 column 3:0: expected "127.0.0.3" but was "127.0.0.1"
row 1 column 4:a list containing
row 1 column 4:0: expected "gamma" but was "beta"
row 2 column 3:a list containing
row 2 column 3:0: expected "127.0.0.1" but was "127.0.0.3"
row 2 column 4:a list containing
row 2 column 4:0: expected "epsilon" but was "gamma"
```

**Issue Reasons:**
- [main] 2 failures in test test {stats.MaxOfIpGrouping} (20.0% fail rate in 10 executions)
- [main] 2 failures in step 8.16.0_bwc-snapshots (22.2% fail rate in 9 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -103,9 +103,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.spatial.SpatialPushDownGeoPointIT
   method: testPushedDownQueriesSingleValue
   issue: https://github.com/elastic/elasticsearch/issues/111084
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {stats.MaxOfIpGrouping}
-  issue: https://github.com/elastic/elasticsearch/issues/111107
 - class: org.elasticsearch.ingest.geoip.EnterpriseGeoIpDownloaderIT
   method: testEnterpriseDownloaderTask
   issue: https://github.com/elastic/elasticsearch/issues/111002
@@ -114,9 +111,6 @@ tests:
   issue: https://github.com/elastic/elasticsearch/issues/110982
 - class: org.elasticsearch.multi_node.GlobalCheckpointSyncActionIT
   issue: https://github.com/elastic/elasticsearch/issues/111124
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {stats.MinOfIpGrouping}
-  issue: https://github.com/elastic/elasticsearch/issues/111125
 - class: org.elasticsearch.xpack.security.authz.store.NativePrivilegeStoreCacheTests
   method: testGetPrivilegesUsesCache
   issue: https://github.com/elastic/elasticsearch/issues/110788
@@ -126,15 +120,6 @@ tests:
 - class: org.elasticsearch.cluster.PrevalidateShardPathIT
   method: testCheckShards
   issue: https://github.com/elastic/elasticsearch/issues/111134
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {stats.MaxOfIpGrouping SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/111136
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {stats.MinOfIpGrouping SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/111137
-- class: org.elasticsearch.xpack.esql.qa.multi_node.EsqlSpecIT
-  method: test {stats.MinOfIpGrouping ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/111138
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test021InstallPlugin
   issue: https://github.com/elastic/elasticsearch/issues/110343

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
@@ -67,12 +67,13 @@ required_capability: agg_max_min_ip_support
 from hosts
 | eval x = ip0
 | where host > "alpha"
-| stats max(ip0), a = max(ip0), b = max(x), c = max(case(host == "beta", ip0, ip1)) by host;
+| stats max(ip0), a = max(ip0), b = max(x), c = max(case(host == "beta", ip0, ip1)) by host
+| sort host asc;
 
 max(ip0):ip | a:ip | b:ip | c:ip | host:keyword
 127.0.0.1 | 127.0.0.1 | 127.0.0.1 | 127.0.0.1 | beta
-fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | fe81::cae2:65ff:fece:feb9 | gamma
 fe82::cae2:65ff:fece:fec0 | fe82::cae2:65ff:fece:fec0 | fe82::cae2:65ff:fece:fec0 | fe82::cae2:65ff:fece:fec0 | epsilon
+fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | fe81::cae2:65ff:fece:feb9 | gamma
 ;
 
 minOfBooleanExpression
@@ -111,12 +112,13 @@ required_capability: agg_max_min_ip_support
 from hosts
 | eval x = ip0
 | where host > "alpha"
-| stats min(ip0), a = min(ip0), b = min(x), c = min(case(host == "beta", ip0, ip1)) by host;
+| stats min(ip0), a = min(ip0), b = min(x), c = min(case(host == "beta", ip0, ip1)) by host
+| sort host asc;
 
 min(ip0):ip | a:ip | b:ip | c:ip | host:keyword
 127.0.0.1 | 127.0.0.1 | 127.0.0.1 | 127.0.0.1 | beta
-fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | 127.0.0.3 | gamma
 fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | 127.0.0.1 | epsilon
+fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | fe80::cae2:65ff:fece:feb9 | 127.0.0.3 | gamma
 ;
 
 maxOfShort

